include/ginkgo/core/log/papi.hpp

```diff
@@ -209,8 +209,9 @@ class Papi : public Logger {
            const Logger::mask_type& enabled_events = Logger::all_events_mask)
     {
         return std::shared_ptr<Papi>(new Papi(enabled_events), [](auto logger) {
-            papi_sde_shutdown(logger->get_handle());
+            auto handle = logger->get_handle();
             delete logger;
+            papi_sde_shutdown(handle);
```

Comment:
This should be calling the other `create` function

Reply:
Do you mean the same fix for the deprecated `create`? I have done it.
Or, you mean different things

Reply:
I mean there should only be a single place this needs to be fixed, the deprecated version should call the normal version instead of reimplementing it. (also it's probably more appropriate to put this into the source file